packages/flutter/test/material/expansion_panel_test.dart

```diff
@@ -1255,4 +1262,40 @@ void main() {
     expect(find.text('C'), findsOneWidget);
     expect(find.text('D'), findsNothing);
   });
+
+  testWidgets('Correct custom expanded padding', (WidgetTester tester) async {
+    const Key firstPanelKey = Key('firstPanelKey');
+
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: SingleChildScrollView(
+          child: SimpleExpansionPanelListTestWidget(
+            firstPanelKey: firstPanelKey,
+            canTapOnHeader: true,
+            expandedHeaderPadding: EdgeInsets.symmetric(vertical: 40.0),
+          ),
+        ),
+      ),
+    );
+
+    // The panel is closed
+    expect(find.text('A'), findsOneWidget);
+    expect(find.text('B'), findsNothing);
+    RenderBox box = tester.renderObject(find.byType(ExpansionPanelList));
+    final double oldHeight = box.size.height;
+    expect(box.size.height, equals(oldHeight));
+
+    // Now, expand the child panel.
+    await tester.tap(find.byKey(firstPanelKey));
+    await tester.pumpAndSettle();
+
+    // The panel is expanded
+    expect(find.text('A'), findsNothing);
+    expect(find.text('B'), findsOneWidget);
+    box = tester.renderObject(find.byType(ExpansionPanelList));
+
+    // Padding is added to top and bottom of expanded header (subtracting body height)
+    final double heightDelta = box.size.height - oldHeight - 107.0;
```

Comment:
Where does `107.0` come from? 

I think it would be better to measure the distance between the edges of the expansion panel's header from its contents instead of doing a comparison between the height when it was closed with the height when it is open, since that doesn't really say too much about the actual padding you're expecting. Plus, it creates a dependency on the original padding/size being correct when checking the expanded padding. 

This will require some digging into the widgets that make up the expansion panel header and using [Finders](https://master-api.flutter.dev/flutter/flutter_test/Finder-class.html) to then get the render objects of the components of the expansion panel header.

Reply:
Good idea testing only the exact header box. Please have a look at the latest draft if the best finder approeach is used.